pkg/app/rds_aurora_mysql.go

```
@@ -0,0 +1,432 @@
+		return errors.Wrapf(err, "Error getting aws config app=%s", a.name)
+	}
+
+	// Create ec2 client
+	ec2Cli, err := ec2.NewClient(ctx, awsConfig, region)
+	if err != nil {
+		return err
+	}
+
+	// Create security group
+	log.Info().Print("Creating security group.", field.M{"app": a.name, "name": a.securityGroupName})
+	sg, err := ec2Cli.CreateSecurityGroup(ctx, a.securityGroupName, "To allow ingress to Aurora DB cluster")
+	if err != nil {
+		return errors.Wrap(err, "Error creating security group")
+	}
+	a.securityGroupID = *sg.GroupId
+
+	// Add ingress rule
+	_, err = ec2Cli.AuthorizeSecurityGroupIngress(ctx, a.securityGroupName, "0.0.0.0/0", "tcp", 3306)
+	if err != nil {
+		return errors.Wrap(err, "Error authorizing security group")
+	}
+
+	rdsCli, err := rds.NewClient(ctx, awsConfig, region)
+	if err != nil {
+		return err
+	}
+
+	// Create RDS instance
+	log.Info().Print("Creating RDS Aurora DB cluster.", field.M{"app": a.name, "id": a.id})
+	_, err = rdsCli.CreateDBCluster(ctx, AuroraDBStorage, AuroraDBInstanceClass, a.id, string(function.DBEngineAuroraMySQL), a.dbName, a.username, a.password, []string{a.securityGroupID})
+	if err != nil {
+		return errors.Wrap(err, "Error creating DB cluster")
+	}
+
+	err = rdsCli.WaitUntilDBClusterAvailable(ctx, a.id)
+	if err != nil {
+		return errors.Wrap(err, "Error waiting for DB cluster to be available")
+	}
+
+	// create db instance in the cluster
+	_, err = rdsCli.CreateDBInstanceInCluster(ctx, a.id, fmt.Sprintf("%s-instance-1", a.id), AuroraDBInstanceClass, string(function.DBEngineAuroraMySQL))
+	if err != nil {
+		return errors.Wrap(err, "Error creating an instance in Aurora DB cluster")
+	}
+
+	err = rdsCli.WaitUntilDBInstanceAvailable(ctx, fmt.Sprintf("%s-instance-1", a.id))
+	if err != nil {
+		return errors.Wrap(err, "Error waiting for DB instance to be available")
+	}
+
+	dbCluster, err := rdsCli.DescribeDBClusters(ctx, a.id)
+	if err != nil {
+		return err
+	}
+	a.host = *dbCluster.DBClusters[0].Endpoint
```

This would crash if DBCluster empty. We should verify the len != 0 and fail early

---

I don't think this would be empty if `DescribeDBClusters` line 172 was successful, but let's add this.